src/plugins/tangorin.rs

```diff
@@ -0,0 +1,186 @@
+            None => None
+        }
+    }
+
+    fn retrieve_from_selector(&self, doc: &kuchiki::NodeRef, selector: &str) -> Option<String> {
+        doc.select(selector).unwrap().next().map(|match_| {
+                let node = match_.as_node().first_child().unwrap();
+                let borrowed_text = node.as_text().unwrap().borrow();
+                borrowed_text.to_owned().trim().to_string()
+        })
+    }
+
+    fn retrieve_meaning(&self, doc: &kuchiki::NodeRef) -> Option<String> {
+        doc.select("span[class=eng]").unwrap().next().map(|match_| {
+            let node_children = match_.as_node().children();
+            let mut meaning = String::new();
+            for child in node_children {
+                if let Some(text) = child.as_text() {
+                    meaning.push_str(text.borrow().as_str());
+                }
+                else {
+                    if let Some(text) = self.inner_text(&child) {
+                        meaning.push_str(text.as_str());
+                    }
+                }
+            }
+
+            meaning
+        })
+    }
+
+    fn retrieve_info(&self, doc: &kuchiki::NodeRef) -> Option<String> {
+        match doc.select("span[class=eng]").unwrap().next() {
+            None => None,
+            Some(match_) => {
+                let node = match_.as_node();
+                let following_siblings = node.following_siblings();
+
+                match following_siblings.select("i[class=d-info]") {
+                    Err(_) => None,
+                    Ok(mut info_sibs) => info_sibs.next().map(|first_sibling|
+                        self.inner_text(first_sibling.as_node()).unwrap()
+                    )
+                }
+            }
+        }
+    }
+
+    fn inner_text(&self, root: &kuchiki::NodeRef) -> Option<String> {
+        match root.first_child(){
+            None => None,
+            Some(match_) => {
+                match match_.as_text() {
+                    Some(result_str) => Some(result_str.borrow().to_owned()),
+                    None => None
+                }
+            }
+        }
+    }
+
+    fn tangorin(&self, server: &IrcServer, _: &Message, target: &str, msg: &str) -> io::Result<()> {
+        let url = match self.grep_kanji(msg) {
+            Some(kanji) => format!("http://tangorin.com/general/{}", kanji),
+            None      => { return Ok(()); }
+        };
+
+        if let Ok(doc) = kuchiki::parse_html().from_http(&url) {
+            let romaji = try_option!(self.retrieve_from_selector(&doc, "rt"));
+            let kana = try_option!(self.retrieve_from_selector(&doc, "rb"));
+            let kanji = try_option!(self.retrieve_from_selector(&doc, "span[class=writing]"));
+            let meaning = try_option!(self.retrieve_meaning(&doc));
+            let info : String = match self.retrieve_info(&doc) {
+                Some(retrieved) => format!(" ({})", retrieved.replace("\u{2014}", "").replace(".", "").to_lowercase()),
```

Comment:
unfortunately [String::remove](https://doc.rust-lang.org/std/string/struct.String.html#method.remove) does not do what we thought it does: 
`Removes a char from this String at a byte position and returns it.`.
String::replace is still the way to go.